client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -192,4 +200,50 @@ final case class Client(open: Service[Request, DisposableResponse], shutdown: Ta
     shutdown.run
 }
 
+<<<<<<< HEAD
+object Client {
+  /** Creates a mock client from the specified service.  Useful for generating
+    * pre-determined responses for requests in testing.
+    * 
+    * @param service the service to respond to requests to this client
+    */
+  def mock(service: HttpService): Client = {
```

Comment:
I don't particularly like placing test related code in the main package. Would it be better to make a new subproject for testing related things? I doubt people using mocks are doing so in any other place than testing, so keeping it in a subproject would keep it off their production JVM as well.